internal/uvm/virtual_device.go

```diff
@@ -0,0 +1,47 @@
+package uvm
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Microsoft/go-winio/pkg/guid"
+	"github.com/Microsoft/hcsshim/internal/guestrequest"
+	"github.com/Microsoft/hcsshim/internal/requesttype"
+	hcsschema "github.com/Microsoft/hcsshim/internal/schema2"
+)
+
+const (
+	resourcePathVPCIFmt = "VirtualMachine/Devices/VirtualPci/%s"
+)
+
+func (uvm *UtilityVM) AssignDevice(ctx context.Context, device hcsschema.VirtualPciDevice) (string, error) {
+	guid, err := guid.NewV4()
+	if err != nil {
+		return "", err
+	}
+	id := guid.String()
```

Comment:
Pretty sure if you pass a value to `Sprintf` with a `%s` format specifier, it will automatically call `.String()`, so we shouldn't need to do that here.

Reply:
I still need a string of the guid below for the VMBusGUID setting sent for the guest side request